packages/govuk-frontend/src/govuk/common/configuration.mjs

```diff
@@ -14,6 +16,27 @@ import { isObject, formatErrorMessage } from './index.mjs'
  * @augments GOVUKFrontendComponent<RootElementType>
  */
 export class GOVUKFrontendComponentConfigurable extends GOVUKFrontendComponent {
+  /**
+   * configOverride
+   *
+   * A function that be defined by a component that extends
+   * this class. Intended usage is it takes as a parameter a
```

Comment:
This is all quite wordy. Maybe something like:

Function which overrides configuration to prioritize properties from the root element's dataset.

It should take a subset of configuration as input and return a new configuration object with properties that should be overridden based on the root element's dataset. A Symbol is used for indexing to prevent conflicts.

Reply:
That's a much better phrasing! I found it quite challenging to describe the definition of the virtual function.

Reply:
Added that wording!